src/core/reactor.cc

```diff
@@ -1137,6 +1137,8 @@ void cpu_stall_detector::on_signal() {
         // no task was processed - report unless supressed
         maybe_report();
         _report_at <<= 1;
+        // Both lowres clocks could become outdated after a reactor stall.
+        lowres_clock::update();
```

Comment:
The timers are updated outside of the reporting code. It will be performed on every timer tick (if the task is the same).

I understand that there are plenty of ways to break the invariant with reactor stalls. But this case is very difficult to debug. It's not possible to fix in general case but this change will make it less likely to appear. 

In my case it was difficult to debug because the error manifested as a log message that was saying that the operation timed out and initial time budget was 0ms. This looked like some other bug, likely a memory bug that caused the variable that contained a time budget to be overwritten or a lifetime issue. The clock is still monophonic though. So it doesn't violate any important invariants.

Reply:
The code which prints the time budget, was it after a preemption point inside some_io_operation()? If so, then you can still see such messages.